### PR TITLE
added IDs to requested components

### PR DESCRIPTION
**Pull request by danielfmiranda**
# Description

Related PRs/issues: #9609

This PR adds the following IDs to the components requested in #9609:

| Location      | ID |  Selector |
| ----------- | ----------- |-----------|
| Featured Article Div      | `featured-article`       | `#featured-article`  |
| Related Article Section within Featured Article DIv   | `supporting-articles`        | `#supporting-articles`  |
| Article Page Related Articles (Top)  | `article-primary-related-articles`        | `#article-primary-related-articles`  |
| Article Page Related Articles (Bottom)  | `article-secondary-related-articles`        | `#article-secondary-related-articles` |
| Article Index Page Related Articles  | `article-index-popular-articles`        | `#article-index-popular-articles` |
| Product Page Related Articles (Top)  | `product-primary-related-articles`        | `#product-primary-related-articles` |
| Product Page Related Articles (Bottom)  | `product-secondary-related-articles` | `#product-secondary-related-articles` |
| Category Page Related Articles (Top)  | `category-primary-related-articles` | `#category-primary-related-articles` |
| Category Page Related Articles (Bottom)  | `category-secondary-related-articles` | `#category-secondary-related-articles` |
| Product Page's Related Products  | `related-products` | `#related-products` |
| Newsletter sign up form in the product grid  | `product-grid-newsletter-signup` | `#product-grid-newsletter-signup` |
| Newsletter sign up form in Article Index Page  | `article-index-newsletter-signup` | `#article-index-newsletter-signup`|
| Newsletter sign up form in Article Pages| `article-newsletter-signup` | `#article-newsletter-signup`|

# Checklist

**Tests**
- ~~[ ] Is the code I'm adding covered by tests?~~

**Changes in Models:**
- ~~[ ] Did I update or add new fake data?~~
- ~~[ ] Did I squash my migration?~~
- ~~[ ] Are my changes [backward-compatible](https://github.com/mozilla/foundation.mozilla.org/blob/main/docs/workflow.md#django-migrations-what-to-do-when-working-on-backward-incompatible-migrations). If not, did I schedule a deploy with the rest of the team?~~

**Documentation:**
- ~~[ ] Is my code documented?~~
- ~~[ ] Did I update the READMEs or wagtail documentation?~~

### CODE REVIEW

**Review by nancyt1 — APPROVED** [review]
This is great! Thank you!

**Review by tbrlpld — APPROVED** [review]
Thanks @danielfmiranda Just two small optional questions.

File: network-api/networkapi/templates/pages/buyersguide/product_page.html [hunk 1]
```diff
@@ -109,7 +109,9 @@ <h1 class="tw-h1-heading col-12">{{product.title}}</h1>
 
               {% with primary_related_articles=product.get_primary_related_articles  %}
                 {% if primary_related_articles %}
+                <div id="product-primary-related-articles">
                   {% include "fragments/buyersguide/related_reading.html" with articles=primary_related_articles %}
+                </div>
```
[the remark this answers is not in the document]
Comment (danielfmiranda) [reply network-api/networkapi/templates/pages/buyersguide/product_page.html:114]: Updated!

File: network-api/networkapi/templates/fragments/buyersguide/featured_article.html [hunk 2]
```diff
@@ -50,7 +50,7 @@ <h2 class="tw-h3-heading group-hover:tw-underline">
     </a>
 
     {% if supporting_articles %}
-      <div class="tw-col-span-full medium:tw-col-span-6 large:tw-col-span-5">
+      <div id="related-articles" class="tw-col-span-full medium:tw-col-span-6 large:tw-col-span-5">
```
[the remark this answers is not in the document]
Comment (danielfmiranda) [reply network-api/networkapi/templates/fragments/buyersguide/featured_article.html:53]: Sounds good to me, updated

**Comment by danielfmiranda** [comment]
Thanks for the feedback Tibor, both suggestions made sense to me so they have been added!

**Pull request merged**